.github/workflows/build.yaml

```diff
@@ -26,7 +26,37 @@ jobs:
       run: cargo clippy --all-targets --all-features -- -D warnings
 
     - name: 🛠️ Build
-      run: cargo build --verbose
-
+      run: |
+        if [[ ${{ github.ref }} == "refs/heads/main" ]]; then
+          cargo build --verbose --release
+        else
+          cargo build --verbose
+        fi
+     
     - name: 🧪 Test
       run: cargo test --verbose
+
+    - name: 📦 Package
+      run: | 
+          cargo install cargo-deb
+          cargo deb
+
+    - name: 📤 Upload 
+      env:
+        ACCESS_KEY_ID: ${{ secrets.ACCESS_KEY_ID }}
+        SECRET_ACCESS_KEY: ${{ secrets.SECRET_ACCESS_KEY }}
+      run: |
+          sudo apt-get update
+          sudo apt-get install -y rubygems locales
+          sudo gem install deb-s3
+          if [[ ${{ github.ref }} == "refs/heads/main" ]]; then
+            RELEASE=stable
+          else
+            RELEASE=unstable
+          fi
```

Comment:
I wonder if we should not be more conservative, i.e. do unstable on push to `main` and `stable` when we do release (e.g. we create a tag). Otherwise we will have deployment on each commit pushed to the branch that wants to merge to `main` which may be a bit too much.

Reply:
yes good idea!